lib/DDG/Spice/JustDeleteMe.pm

```diff
@@ -0,0 +1,48 @@
+package DDG::Spice::JustDeleteMe;
+
+# ABSTRACT: Search JustDelete.Me for the web service
+
+use DDG::Spice;
+spice is_cached => 1;
+spice wrap_jsonp_callback => 1;
+spice to => 'https://raw.githubusercontent.com/rmlewisuk/justdelete.me/master/sites.json';
+
+my $MIN_QUERY_LENGTH = 4;
+my @tr = qw (delete cancel remove);
+my @pred = qw (my on how to howto can i one his her their theirs a an permanently from of);
+my @req = qw (account accounts);
+triggers any => @tr;
+
+handle query_parts => sub {
+    my $query = validate(@_);
+    if (length($query) < $MIN_QUERY_LENGTH) {
+        return;
+    }
+    else {
+        return $query;
+    }
+};
+
+sub validate {
```

Comment:
I suspect we can probably replace this functionality with one or two regular expressions?

Something like this should cover most of the queries?

```perl
/how (to|can|can (i|one))? (delete|remove|cancel) (his|her|their)? (\w+) accounts? (permanently)?/;
```

Note: I haven't tested this, but I suggest you look at the autocomplete suggestions to see which forms of queries are most popular "delete X account" seems very common :+1: